File: apps/web/repositories/api/automation.repository.ts
// API Automation Repository implementation
import type { IAutomationRepository } from '../types';
import type { AutomationOut, AutomationCreate, AutomationUpdate, AutomationRunResult } from '@/types/api';
import { apiClient } from '@/services/api-client';

export class ApiAutomationRepository implements IAutomationRepository {
    public async list(homeId?: string): Promise<AutomationOut[]> {
        return apiClient.get('/automations', {
            params: homeId ? { home_id: homeId } : undefined,
        });
    }

    public async create(req: AutomationCreate): Promise<AutomationOut> {
        return apiClient.post('/automations', req);
    }

    public async get(id: string): Promise<AutomationOut> {
        return apiClient.get(`/automations/${id}`);
    }

    public async update(id: string, req: AutomationUpdate): Promise<AutomationOut> {
        return apiClient.patch(`/automations/${id}`, req);
    }

    public async delete(id: string): Promise<void> {
        return apiClient.delete(`/automations/${id}`);
    }

    public async trigger(id: string, context?: Record<string, any>): Promise<AutomationRunResult> {
        return apiClient.post(`/automations/${id}/trigger`, context || {});
    }
}

File: apps/android/core/src/main/java/com/atech/core/data/AutomationRepository.kt
package com.atech.core.data

import com.atech.core.common.DomusResult
import com.atech.core.model.Automation
import com.atech.core.model.AutomationCreate
import com.atech.core.model.AutomationRunResult
import com.atech.core.model.AutomationUpdate
import com.atech.core.network.DomusHttp
import com.atech.core.network.safeApiCall
import io.ktor.client.request.delete
import io.ktor.client.request.get
import io.ktor.client.request.parameter
import io.ktor.client.request.patch
import io.ktor.client.request.post
import io.ktor.client.request.setBody

/** Automations. Mirrors `backend/automations/router.py`. */
class AutomationRepository(private val http: DomusHttp) {
    private val base get() = "${http.config.apiBase}/automations"

    suspend fun list(homeId: String? = null): DomusResult<List<Automation>> =
        safeApiCall { http.client.get(base) { parameter("home_id", homeId) } }

    suspend fun get(automationId: String): DomusResult<Automation> =
        safeApiCall { http.client.get("$base/$automationId") }

    suspend fun create(data: AutomationCreate): DomusResult<Automation> =
        safeApiCall { http.client.post(base) { setBody(data) } }

    suspend fun update(automationId: String, data: AutomationUpdate): DomusResult<Automation> =
        safeApiCall { http.client.patch("$base/$automationId") { setBody(data) } }

    suspend fun delete(automationId: String): DomusResult<Unit> =
        safeApiCall { http.client.delete("$base/$automationId") }

    /** Manually fire an automation (evaluates conditions, runs actions). */
    suspend fun trigger(automationId: String): DomusResult<AutomationRunResult> =
        safeApiCall { http.client.post("$base/$automationId/trigger") }
}
